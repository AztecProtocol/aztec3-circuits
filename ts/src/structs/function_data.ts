import { serializeToBuffer } from "../wasm/serialize.js";

/**
 * @see abis/function_data.hpp
 */
export class FunctionData {
  constructor(
    public functionSelector: number,
    public isPrivate: true,
    public isConstructor: true
  ) {}
  toBuffer(): Buffer {
    return serializeToBuffer(
      this.functionSelector,
      this.isPrivate,
      this.isConstructor
    );
  }
}
