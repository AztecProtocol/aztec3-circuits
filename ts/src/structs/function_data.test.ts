import { expectSerializeToMatchSnapshot } from "../tests/expectSerializeToMatchSnapshot.js";
import { FunctionData } from "./function_data.js";

describe("basic FunctionData serialization", () => {
  it(`serializes a trivial FunctionData and prints it`, async () => {
    // Test the data case: writing (mostly) sequential numbers
    for (let i = 0; i < 16; i++) {
      await expectSerializeToMatchSnapshot(
        new FunctionData(1, true, true).toBuffer(),
        "abis__test_roundtrip_serialize_function_data"
      );
    }
  });
});
