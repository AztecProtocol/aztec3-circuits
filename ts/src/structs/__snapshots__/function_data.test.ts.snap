// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic FunctionData serialization serializes a trivial FunctionData and prints it 1`] = `
"function_selector: 0
is_private: 1
is_constructor: 1
"
`;

exports[`basic FunctionData serialization serializes a trivial FunctionData and prints it 2`] = `
"function_selector: 16777216
is_private: 1
is_constructor: 1
"
`;

exports[`basic FunctionData serialization serializes a trivial FunctionData and prints it 3`] = `
"function_selector: 33554432
is_private: 1
is_constructor: 1
"
`;

exports[`basic FunctionData serialization serializes a trivial FunctionData and prints it 4`] = `
"function_selector: 50331648
is_private: 1
is_constructor: 1
"
`;

exports[`basic FunctionData serialization serializes a trivial FunctionData and prints it 5`] = `
"function_selector: 67108864
is_private: 1
is_constructor: 1
"
`;

exports[`basic FunctionData serialization serializes a trivial FunctionData and prints it 6`] = `
"function_selector: 83886080
is_private: 1
is_constructor: 1
"
`;

exports[`basic FunctionData serialization serializes a trivial FunctionData and prints it 7`] = `
"function_selector: 100663296
is_private: 1
is_constructor: 1
"
`;

exports[`basic FunctionData serialization serializes a trivial FunctionData and prints it 8`] = `
"function_selector: 117440512
is_private: 1
is_constructor: 1
"
`;

exports[`basic FunctionData serialization serializes a trivial FunctionData and prints it 9`] = `
"function_selector: 134217728
is_private: 1
is_constructor: 1
"
`;

exports[`basic FunctionData serialization serializes a trivial FunctionData and prints it 10`] = `
"function_selector: 150994944
is_private: 1
is_constructor: 1
"
`;

exports[`basic FunctionData serialization serializes a trivial FunctionData and prints it 11`] = `
"function_selector: 167772160
is_private: 1
is_constructor: 1
"
`;

exports[`basic FunctionData serialization serializes a trivial FunctionData and prints it 12`] = `
"function_selector: 184549376
is_private: 1
is_constructor: 1
"
`;

exports[`basic FunctionData serialization serializes a trivial FunctionData and prints it 13`] = `
"function_selector: 201326592
is_private: 1
is_constructor: 1
"
`;

exports[`basic FunctionData serialization serializes a trivial FunctionData and prints it 14`] = `
"function_selector: 218103808
is_private: 1
is_constructor: 1
"
`;

exports[`basic FunctionData serialization serializes a trivial FunctionData and prints it 15`] = `
"function_selector: 234881024
is_private: 1
is_constructor: 1
"
`;

exports[`basic FunctionData serialization serializes a trivial FunctionData and prints it 16`] = `
"function_selector: 251658240
is_private: 1
is_constructor: 1
"
`;
